src/System.Text.Json/src/System/Text/Json/ThrowHelper.cs

```diff
@@ -461,7 +461,14 @@ private static string GetResourceString(ExceptionResource resource, int currentD
             {
                 case ExceptionResource.MismatchedObjectArray:
                     Debug.Assert(token == JsonConstants.CloseBracket || token == JsonConstants.CloseBrace);
-                    message = SR.Format(SR.MismatchedObjectArray, (char)token);
+                    if (tokenType == JsonTokenType.PropertyName)
+                    {
+                        message = SR.Format(SR.CannotWriteEndAfterProperty, (char)token);
+                    }
+                    else
+                    {
+                        message = SR.Format(SR.MismatchedObjectArray, (char)token);
+                    }
```

Comment:
Nit:
```C#
message = (tokenType == JsonTokenType.PropertyName) ?
    SR.Format(SR.CannotWriteEndAfterProperty, (char)token) :
    SR.Format(SR.MismatchedObjectArray, (char)token);
```
?  That's just personal preference, so if you don't want to change it, that's fine.  I just prefer this syntax in cases like this because it clearly highlights the difference, factoring out the `message = ` part that's the same across them.